### PR TITLE
Issue-606 Enhancements in developer environment

### PR DESCRIPTION
**Pull request by eolivelli**
Improvements in maven configuration in order to speed up development

### CODE REVIEW

**Comment by eolivelli** [comment]
see #660 for the list of improvements

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/bookkeeper-precommit-pullrequest-docker/145/

File: bookkeeper-server/pom.xml [hunk 1]
```diff
@@ -435,6 +436,58 @@
...
+               <configuration>
+                   <skip>true</skip>
+               </configuration>
+            </plugin>
+            <plugin>
+              <groupId>org.apache.maven.plugins</groupId>
+              <artifactId>maven-surefire-plugin</artifactId>
+              <version>${maven-surefire-plugin.version}</version>
+              <configuration>
+                <argLine>-Xmx2G -Djava.net.preferIPv4Stack=true -Dlog4j.configuration=log4j_debug.properties</argLine>
```
Comment (ivankelly) [inline bookkeeper-server/pom.xml]: don't change the log configuration property. Rather, change the log4j.properties file to set:
bookkeeper.root.logger=INFO,CONSOLE
log4j.rootLogger=${bookkeeper.root.logger}

Then the dev profile can override this using -Dbookkeeper.root.logger=DEBUG,CONSOLE

Comment (eolivelli) [reply bookkeeper-server/pom.xml]: I will try, I am not an expert of log4j, thank you @ivankelly

**Comment by eolivelli** [comment]
@ivankelly updated the patch with your suggestion
it works like a charm !

**Comment by eolivelli** [comment]
@ivankelly do you have time merge this one ?

**Comment by ivankelly** [comment]
@eolivelli will do, once the checks complete.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/bookkeeper-precommit-pullrequest-docker/146/

**Comment by ivankelly** [comment]
Failure is CompactionTest, which seems to be a clear winner in the flakey league. Merging.

**Pull request closed without merging**